File: app/src/main/java/antoninovitale/dropcodechallenge/list/BeerListView.java
package antoninovitale.dropcodechallenge.list;

import java.util.List;

import antoninovitale.dropcodechallenge.list.model.IBeerListModel;

/**
 * Created by antoninovitale on 02/09/2017.
 */
public interface BeerListView {
    void setRefreshing(boolean value);

    void refreshList();

    void setItems(List<IBeerListModel> models);

    void selectItem(int position);

    void navigateToDetails();

    void showError();
}

File: app/src/main/java/antoninovitale/punkapi/app/list/BeerListView.kt
package antoninovitale.punkapi.app.list

import antoninovitale.punkapi.app.list.model.IBeerListModel

/**
 * Created by antoninovitale on 02/09/2017.
 */
interface BeerListView {
    fun setRefreshing(value: Boolean)
    fun refreshList()
    fun setItems(models: List<IBeerListModel?>?)
    fun selectItem(position: Int)
    fun navigateToDetails()
    fun showError()
}